### PR TITLE
Switch to IDictionary objects for public functions

### PR DESCRIPTION
**Pull request by averche**
## Description

`IDictionary` is a public interface for a `Dictionary` which does not assume a specific internal implementation, which seems better suited for our public functions.

## How has this been tested?

```sh
make regen && make build
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -245,7 +245,7 @@ Each generic operation has a synchronous and asynchronous version.
 ```csharp
 // Generic read from a path with query parameters
 var readPath = "/some/path"
-Dictionary<string, object> queryParams = new Dictionary<string, object>
+IDictionary<string, object> queryParams = new Dictionary<string, object>
```
Comment (AnPucel) [inline README.md:248]: I'd like to keep the examples as simple as possible with the concrete implementation, but I suppose it shows that the methods accept an `IDictionary` now, which makes sense

**Review by AnPucel — APPROVED** [review]
Works for me

File: generate/templates/libraries/httpclient/ApiClient.mustache [hunk 2]
```diff
@@ -252,7 +252,7 @@ namespace {{packageName}}.Client
         /// <summary>
         /// Adds a dictionary of custom headers to current list of custom headers.
         /// </summary>
-        internal void AddCustomHeaders(Dictionary<string, string> headersToAdd)
+        internal void AddCustomHeaders(IDictionary<string, string> headersToAdd)
```
Comment (AnPucel) [inline generate/templates/libraries/httpclient/ApiClient.mustache:255]: I think there's a custom headers example as well, that should probably be updated

File: README.md (hunk 1 above)
Comment (averche) [reply README.md:248]: Yes, I figured that's the typical way you would indicate it but I'm OK using concrete dictionary or even in-lining this param for the example.

File: generate/templates/libraries/httpclient/ApiClient.mustache (hunk 2 above)
Comment (averche) [reply generate/templates/libraries/httpclient/ApiClient.mustache:255]: I've updated the example

**Pull request merged**